x/ccv/provider/keeper/throttle.go

```diff
@@ -102,7 +102,9 @@ func (k Keeper) GetSlashMeterAllowance(ctx sdktypes.Context) math.Int {
 
 	// Compute allowance in units of tendermint voting power (integer),
 	// noting that total power changes over time
-	totalPower := k.stakingKeeper.GetLastTotalPower(ctx)
+
+	// Get total total power of the last consensus validator set on the provider
+	totalPower := k.GetLastTotalProviderConsensusPower(ctx)
```

Comment:
Note that this changes the behavior. `stakingKeeper.GetLastTotalPower` gets the total power from the store, while `GetLastTotalProviderConsensusPower` iterates over 180 keys and recomputes the power every time.